juriscraper/opinions/united_states/state/nyappterm_1st.py

```diff
@@ -3,31 +3,56 @@
 # Court Short Name: NY
 
 import re
-from datetime import date, timedelta
-from typing import Any, Dict
+from datetime import date, datetime, timedelta
+from typing import Any, Dict, Optional, Tuple
 
+from juriscraper.AbstractSite import logger
+from juriscraper.lib.date_utils import make_date_range_tuples
 from juriscraper.OpinionSiteLinear import OpinionSiteLinear
 
 
 class Site(OpinionSiteLinear):
+    first_opinion_date = date(2003, 9, 25)
+
+    # If more than 500 results are found, no results will be shown
+    days_interval = 30
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.court = "Appellate Term, 1st Dept"
         self.court_id = self.__module__
         self.url = "https://iapps.courts.state.ny.us/lawReporting/Search?searchType=opinion"
         self._set_parameters()
         self.expected_content_types = ["application/pdf", "text/html"]
+        self.make_backscrape_iterable(kwargs)
+
+    def _set_parameters(
+        self,
+        start_date: Optional[date] = None,
+        end_date: Optional[date] = None,
+    ) -> None:
+        """Set the parameters for the POST request.
+
+        If no start or end dates are given, scrape last month.
+        This is the default behaviour for the present time scraper
+
+        :param start_date:
+        :param end_date:
 
-    def _set_parameters(self) -> None:
-        """Set the parameters for the POST request."""
+        :return: None
+        """
         self.method = "POST"
-        today = date.today()
+
+        if not end_date:
+            start_date = date.today()
+            end_date = start_date - timedelta(days=30)
+
         self.parameters = {
             "rbOpinionMotion": "opinion",
             "Pty": "",
             "and_or": "and",
-            "dtStartDate": (today - timedelta(days=30)).strftime("%m/%d/%Y"),
-            "dtEndDate": today.strftime("%m/%d/%Y"),
+            "dtStartDate": start_date.strftime("%m/%d/%Y"),
```

Comment:
I think there is bug there, if you run the entire scaper for 1st department:

`python sample_caller.py -c juriscraper.opinions.united_states.state.nyappterm_1st
`
is going to post:

 'dtStartDate': '04/27/2024', 'dtEndDate': '03/28/2024'

and that raises a validation exception in the court page: "Start Date must be BEFORE End Date."

I think it should go like this:

```
end_date = date.today()  # e.g. 03/28/2024
start_date = end_date - timedelta(days=30)  # e.g. 04/27/2024
```

Reply:
Thanks for detecting this Kevin, I mixed up the dates